### ISSUE TITLE
Develop a Video-Lesson and Assessment Flow for LMS

### ISSUE THREAD

**Issue by akashreactdeveloper** [continuousactivelearning/cal#80]
## **Description**  
Create a React page for an LMS system that provides a seamless learning experience by:  
1. Embedding YouTube videos sequentially in a vertical scroll layout with no additional video details (e.g., title, channel, suggested videos).  
2. Automatically scrolling to the assessment once the YouTube video finishes.  
3. Displaying the assessment related to the completed video below the video frame.  
4. Allowing users to manually proceed to the next video and assessment using a "Next Task" button.  
5. Using dummy JSON data for now, which can later be replaced with API integration.  
6. Ensuring smooth vertical scrolling, clean UI, and accessibility using ShadCN components.

---

## **Requirements**  

1. **YouTube Video Playback with Hidden Details**  
   - Embed YouTube videos using the iframe player.  
   - Configure the iframe to hide video details like title, channel name, and suggested videos by adding parameters to the URL .  
   - Detect when the YouTube video finishes and automatically scroll to the corresponding assessment section.  
   - Use dummy JSON to load video URLs and metadata (e.g., title, description).  

2. **Assessment Section**  
   - Display the assessment related to the completed video directly below the YouTube frame.  
   - Include various assessment types:  
     - **Multiple-choice questions**: Options with correct answer validation.  
     - **Short answer questions**: Text input for users.  
   - Provide interactive feedback for user responses.  

3. **Manual Navigation**  
   - Provide a "Next Task" button to manually scroll to the next section.  
   - Users can only proceed to the next video and assessment after completing the current assessment.  

4. **Vertical Scroll Flow**  
   - Each video and its assessment should appear in a vertical scroll layout.  
   - Ensure smooth transitions between sections during auto or manual scroll.  

5. **UI and Accessibility**  
   - Use ShadCN components for consistency and style.  
   - Support dark mode with a toggle.  
   - Ensure accessibility (ARIA roles, keyboard navigation, and feedback for all interactions).  

6. **Dummy Data**  
   - Use a JSON object to simulate YouTube video and assessment data.  
   - Ensure the structure is easy to replace with APIs in the future.  

---

## **Acceptance Criteria**  

- **YouTube Video Section**  
  - Videos load in a clean iframe player without showing video titles, channel details, or suggested videos.  
  - Automatically scrolls to the assessment when the video ends.  
  - Users can also manually navigate using the "Next Task" button.  

- **Assessment Section**  
  - Assessments are displayed immediately after their respective videos.  
  - Validations provide immediate feedback for user inputs.  
  - Users must complete the assessment to unlock the next video and assessment.  

- **Flow and Navigation**  
  - Videos and assessments appear sequentially in a vertical scroll layout.  
  - Navigation between tasks is smooth and intuitive.  

- **Dummy Data**  
  - JSON contains video URLs, titles, and related assessment details.  
  - All interactions dynamically update the JSON state.  

- **Performance and Responsiveness**  
  - The page is responsive and works well on desktop and tablets.  
  - Video playback and scrolling transitions are smooth without noticeable lag.  

---

## **Additional Notes**  

- **Hiding Video Details**: Use YouTube's embed parameters  to hide UI details like controls, video info, and related videos.  
- Design should be user-friendly, intuitive, and minimalistic.  
- Handle edge cases, such as:  
  - **No videos available**: Display a message like "No lessons available. Please check back later."  
  - **Unanswered assessment**: Prevent navigation until the user attempts the assessment.  
  - **Failed assessment**: Allow retry for incorrect answers with feedback.  
- Include comments in the code to help developers integrate real API data in the future.  

---

**Comment by KumarSaiUggina** [comment]
Hey @akashreactdeveloper @sidrao2006, Could you please assign this issue to me?

**Comment by rkatara100** [comment]
I want to help you out to solve this issue  if @KumarSaiUggina  complete.Please ask @akashreactdeveloper  the update on this

**Comment by akashreactdeveloper** [comment]
@rkatara100 , We have assigned this task to @KumarSaiUggina, if you also want to solve this issue then you can try, we will consider the best solution from our end.

**Comment by balamanu19** [comment]
Hey @akashreactdeveloper @sidrao2006, Could you please assign this issue to me?

**Comment by sidrao2006** [comment]
@KumarSaiUggina, it has been over a day since this issue was assigned to you. If we do not receive any updates from you today, we may proceed with reassigning the issue.

**Comment by KumarSaiUggina** [comment]
> @KumarSaiUggina, it has been over a day since this issue was assigned to you. If we do not receive any updates from you today, we may proceed with reassigning the issue.

Yes, you can assign others. I will work in parallel. I need some time.

**Comment by sidrao2006** [comment]
This has already been impl.
